artifactory/cli_test.go

```diff
@@ -30,3 +41,135 @@ func TestValidateGoNativeCommand(t *testing.T) {
+		{"withSpec", []string{}, []string{"spec=" + getSpecPath(tests.CopyItemsSpec)}, "${REPO1}/*/", "${REPO2}/", "", "", false},
+		{"withSpecAndPattern", []string{"TestPattern"}, []string{"spec=" + getSpecPath(tests.CopyItemsSpec)}, "", "", "", "", true},
+		{"withPatternTargetAndBuild", []string{"TestPattern", "TestTarget"}, []string{"build=buildName/buildNumber"}, "TestPattern", "", "buildName/buildNumber", "", false},
+		{"withPatternTargetAndBundle", []string{"TestPattern", "TestTarget"}, []string{"bundle=bundleName/bundleVersion"}, "TestPattern", "", "", "bundleName/bundleVersion", false},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			context, buffer := createContext(test.flags, test.args)
+			specFiles, err := prepareCopyMoveCommand(context)
+			if test.expectError {
+				assert.Error(t, err, buffer)
+			} else {
+				assert.NoError(t, err, buffer)
+				assert.Equal(t, test.expectedPattern, specFiles.Get(0).Pattern)
+			}
+		})
+	}
+}
+
+func TestPreparePropsCmd(t *testing.T) {
+	tests := []struct {
+		name            string
+		args            []string
+		flags           []string
+		expectedProps   string
+		expectedPattern string
+		expectedBuild   string
+		expectedBundle  string
+		expectError     bool
+	}{
+		{"withoutPattern", []string{"key1=val1"}, []string{}, "key1=val1", "", "", "", true},
+		{"withPattern", []string{"TestPattern", "key1=val1"}, []string{}, "key1=val1", "TestPattern", "", "", false},
+		{"withBuild", []string{"key1=val1"}, []string{"build=buildName/buildNumber"}, "key1=val1", "*", "buildName/buildNumber", "", false},
+		{"withBundle", []string{"key1=val1"}, []string{"bundle=bundleName/bundleVersion"}, "key1=val1", "*", "", "bundleName/bundleVersion", false},
+		{"withSpec", []string{"key1=val1"}, []string{"spec=" + getSpecPath(tests.SetDeletePropsSpec)}, "key1=val1", "${REPO1}/", "", "", false},
+		{"withSpecAndPattern", []string{"TestPattern", "key1=val1"}, []string{"spec=" + getSpecPath(tests.SetDeletePropsSpec)}, "key1=val1", "", "", "", true},
+		{"withPatternAndBuild", []string{"TestPattern", "key1=val1"}, []string{"build=buildName/buildNumber"}, "key1=val1", "TestPattern", "buildName/buildNumber", "", false},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			context, buffer := createContext(test.flags, test.args)
+			propsCommand, err := preparePropsCmd(context)
+			if test.expectError {
+				assert.Error(t, err, buffer)
+			} else {
+				assert.NoError(t, err, buffer)
+				assert.Equal(t, test.expectedProps, propsCommand.Props())
+				assert.Equal(t, test.expectedPattern, propsCommand.Spec().Get(0).Pattern)
+			}
+		})
+	}
+}
+
+func createContext(testFlags, testArgs []string) (*cli.Context, *bytes.Buffer) {
+	flagSet := flag.NewFlagSet("TestFlagSet", flag.ContinueOnError)
+	flags := append(setStringFlags(flagSet, testFlags...), testArgs...)
+	flagSet.Parse(flags)
```

Comment:
I suggest moving this line into the ```setStringFlags``` function. This should make the flow a bit easier to read.